### DISCUSSION TITLE
Subclustering of integrated dataset

### DISCUSSION THREAD

**Discussion by rythi4** [satijalab/seurat#5214] [Q&A]
I have a SCT integrated dataset, and following integration I would like to subset this dataset and perform clustering on the cells from a single cluster. 
But I am having trouble finding a clear workflow for subclustering, does it exist?
Should I integrate again?

The reason I integrate the entire dataset in the beginning is that if I pull out the cells of interest beforehand, the objects I integrate are very small and this causes some problems during the integration. 

I was thinking to use the following workflow: 
1) Integrate the entire dataset
2) Run dimensionality reduction on the entire dataset (integrated assay)
3) Extract cells of interest
4) Find variable features for the subset using the RNA assay
5) Run dimensionality reduction on the subset using the integrated assay
6) Cluster the cells 

Does it makes sense to do the subclustering like this?

**Comment by cswoboda** [comment]
I don't think there's one best way to do this, it's sort of data dependent. 
Normally I do this:
Steps1-3 similar to you
4. Rerun clustering with the global dimensionality reduction from the first corrected integrated assay. Stop here if things cluster out as expected
5.  If you need more resolution, run it your way, with the unfortunate side effect that FindVariableFeatures could very well pick up on more technical noise than real cell differences that are conserved across datasets, so be wary of this when running your analysis. If you are reruning PCA on the integrated assay though, you're actually not changing anything from the above step I believe, as the features or genes used for that PCA are going to come from your global integration (your anchors), not whatever genes you run with findvariablefeatures. You'd have to run the PCA and the rest in the RNA assay.

**Reply by rythi4** [reply]
Thank you for your input cswoboda

Actually when I rerun PCA, I set the features to equal the variable features I just determined using FindVariableFeatures(). In that case, I don't think it is necessary to switch to the RNA assay?

**Reply by cswoboda** [reply]
Someone can correct me if I'm wrong but I believe that works rather well for your case. Should just be wary in downstream analysis that clusters are meaningful populations and not clustering by replicate or condition due to a variable feature being a batch effect for example. Does that make sense?

**Reply by rythi4** [reply]
Yes that makes sense. Thank you

**Reply by cswoboda** [reply]
Of course, feel free to close if you feel like it's properly answered! If you notice noisy technical genes, you can always take them out. Best of luck!

**Reply by cswoboda** [reply]
Also, have not used this yet, but I believe Seurat V4 has a FindSubClusters() command that may work well in your case. Should read up on that documentation if it might help you
